### PR TITLE
[HEYUI-212] feat/implement dropdown

### PR DESCRIPTION
**Pull request by hcafaq**
## Pull Request
https://heycar.atlassian.net/browse/HEYUI-212
### Description

image

Describe your changes (or add ticket description) and add relevant screenshots

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change

### How do I test this

- Add steps to test
- in bullet point format
- preferably you can add link to the storybook build in the PR

### Did you remember to take care of the following?

- [x] `npm i` – for new NPM dependencies.
- [x] `npm run lint` - to check for linting issues
- [x] `npm run test` - to run unit tests
- [x] `npm run test:sh:docker` - to run screenshot tests, [detail instruction](https://hey-car.github.io/heycar-uikit/main/?path=/docs/guidelines-screenshot-testing--page)

### New Feature / Bug Fix

- [ ] Run unit tests to ensure all existing tests are still passing.
- [ ] Add new passing unit tests to cover the code introduced by your pr.

Thanks for contributing!

[HEYUI-212]: https://heycar.atlassian.net/browse/HEYUI-212?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/dropdown/src/Dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+export const Dropdown = ({ value, onChange, options, disabled, dataTestId }: DropdownProps) => {
+  const [stateValue, setStateValue] = useState<SelectOptions | undefined>(
+    value,
+  );
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  const selectOption = (option: SelectOptions) => {
+    if (onChange) {
```
Comment (esraltintas) [inline packages/dropdown/src/Dropdown.tsx:16]: Hey, @hcafaq could you check all jest coverage tests, please?

Comment (hcafaq) [reply packages/dropdown/src/Dropdown.tsx:16]: resolved

**Comment by hcafaq** [comment]
> When I check with the design seems the icon and text should centre align. Could you check please? @hcafaq
> 
> Screenshot 2023-01-04 at 14 11 43

resolved

**Review by heyabinanto — APPROVED** [review]

**Review by esraltintas — APPROVED** [review]

**Comment by esraltintas** [comment]
@hcafaq Please request to design team to test this new component

**Pull request merged**